### PR TITLE
WebGPURenderer: avoid render pass/pipeline attachment mismatches

### PR DESCRIPTION
**Pull request by aardgoose**
Render pass attachment characteristics and pipeline attachment expectations must match. The following sequence can result in a mismatch followed by WebGPU errors because the chain map keys used to locate matching renderObjects and related pipelines does not represent attachment status.

1. .render( scene, camera ); // pipelines created and keyed by (scene, camera) etc.

2. .setRenderTarget( newTarget );  // new target with mismatch of color format or sample count etc.

3. .render( scene, camera ); // pipelines from (1) used which fail with renderPass for newTarget.

This PR adds the render target to the key for renderContexts and uses renderContext as a replacement key element for renderObjects. This prevents the reuse of incompatible pipelines.

### CODE REVIEW

File: examples/jsm/renderers/common/RenderObjects.js [hunk 1]
```diff
@@ -17,10 +17,10 @@ class RenderObjects {
 
 	}
 
-	get( object, material, scene, camera, lightsNode, passId ) {
+	get( object, material, scene, camera, lightsNode, passId, renderContext ) {
```
Comment (sunag) [inline examples/jsm/renderers/common/RenderObjects.js:20]: I think better this signature: `get( object, material, scene, camera, lightsNode, renderContext, passId )`
`passId` must be opcional.

File: examples/jsm/renderers/common/Renderer.js [hunk 2]
```diff
@@ -87,6 +87,8 @@ class Renderer {
 		this._initialized = false;
 		this._initPromise = null;
 
+		this._nullTarget = {};
```
Comment (sunag) [inline examples/jsm/renderers/common/Renderer.js:90]: I think that we can make static this.

File: examples/jsm/renderers/common/RenderContexts.js [hunk 3]
```diff
@@ -9,9 +9,9 @@ class RenderContexts {
 
 	}
 
-	get( scene, camera ) {
+	get( scene, camera, renderTarget ) {
```
Comment (sunag) [inline examples/jsm/renderers/common/RenderContexts.js:12]: `get( scene, camera, renderTarget = null )` seems better

**Comment by sunag** [comment]
One approach to consider is to cache information of a `RenderTarget` that impacts pipeline and shader creation like `{ colorFormat, sampler }` in order to reduce the numbers of `RenderObjects`.

**Comment by aardgoose** [comment]
I looked at using something like attachmentState = { color format, sample count, depth-stencil } as well, and agree that there is there is more renderObject creation in some cases. I didn't want to add another layer to the renderContext chain map, but that could be avoided by appending attachmentState to the passId string? 

On the 'for' side to adding render target to renderContext is more opportunity for caching the renderPass descriptor and avoiding createView() calls.

**Comment by LeviPesin** [comment]
Maybe we can implement a function like `RenderContext.getCacheKey()` and use that cache key instead of render context?

**Pull request merged**